src/Elastic.Apm/Logging/ScopedLogger.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Concurrent;
+
+namespace Elastic.Apm.Logging
+{
+	internal class ScopedLogger : IApmLogger
+	{
+		public ScopedLogger(IApmLogger logger, string scope) => (Logger, Scope) = (logger, scope);
+
+		private ConcurrentDictionary<string, LogValuesFormatter> Formatters { get; } = new ConcurrentDictionary<string, LogValuesFormatter>();
+
+		public IApmLogger Logger { get; }
+
+		public string Scope { get; }
+
+		public LogValuesFormatter GetOrAddFormatter(string message, int expectedCount)
+		{
+			var formatter = Formatters.GetOrAdd(message, s => new LogValuesFormatter($"{{{{{{Scope}}}}}} {s}", Scope));
```

Comment:
So many curly braces - is it on purpose?

Reply:
Yeah. there is the named `{Scope}` placeholder wrapped in double `{}` so that the end result is `{<scope>}` in the message. The end result is one pair being emitted e.g ` {HttpDiagnosticListener} Exception in OnE....`